meetings/CI-workinggroup/2019-03-06.md

```diff
@@ -0,0 +1,136 @@
+
+* Aloys created Conan repository as an experiment. Conclusion is that we would need to wrap Rez with something else for external package download, whereas Conan supports this out of the box.
+
+* Jeff: Rez is not very viable for ASWF CI builds on cloud systems.
+
+* Aloys: we could provide a set of Rez recipes which are compatible with the build artifacts we generate.
+
+* Dan: what are the ASWF goals with respect to build artifacts? Daniel: CI is the first / foremost goal, but dependency management discussion came up quickly as a key component / requirement of the CI. There is no out of the box solution that everyone agrees on. Every studio has combination of legacy / proprietary tooling, Rez is an industry specific solution to parts of these problems, but doesn’t work all that well with CI / package repository. Dan: could every project build all of its dependencies? Or is sharing dependencies a primary goal? Jeff: not planning to use ASWF artifacts in the short term due to internal requirements. Would love to be able to do so eventually. Daniel: when the LF set up the initial repos they did it as simply as possible, used Linux native packagement to grab dependencies, but quickly ran into the need for dependency management. Aloys: we need guaranteed rather than accidental compatibility, don’t want packages built with slightly different build options (for instance). For instance Autodesk compiles TBB in a specific way, need to make sure you build it the same way, and there’s no well documented build recipe. Sharing those build recipes really helps to make this explicit. 
+
+* Dan: sees build artefacts as mostly targeting the hobbyist developer, large studios will prefer to build everything from source. Can get tricky for builds that depend on quickly changing dependencies like Houdini daily builds for OpenVDB plugins for instance.
+
+* Daniel: challenge from Dan to keep to immediate and tangible goals, especially for projects already in ASWF. OpenVDB currently uses what the OS provides for EXR, Boost (apt-get), but want to better control version of sub components. Need to limit the permutations of the build matrix, one approach is to build with all the latest dependencies, and then do additional build which varies individual components / options.
+
+* Daniel: we need a VFX Reference Platform build of OpenVDB as an experiment. Dan: that should be done on top of CircleCI if we decide to move to that. Larry: maybe not wait after the decision to go to Circle, since that work could inform that decision. Dan: for now stuck in Travis "stages", would rather do that in CircleCI.
+
+* Jeff: what needs to happen next to make the decision on moving to CircleCI? Thanh: we can start using the free tier right away.
+
+* Larry: we will need some kind of estimate of how much cost will we be looking at. Dan: there’s a tool like that for Travis to get how many build minutes have been used for a project.
```

Comment:
(@lgritz @zxiiro @dheckenberg)

There are a few around, but here's one:

http://jsmaniac.github.io/travis-stats/#AcademySoftwareFoundation/openvdb

Very rough estimate from these stats, but OpenVDB appears to average around 200 minutes per build with one or two per day. The sporadic 500 minutes peak is likely when a change causes a full rebuild (ie no ccache). 

Note that we have a nightly cron to assert that nothing in our dependencies has broken - this is expected to succeed, so ideally this should be a low priority build that doesn't use up build minutes. The priority is for pull requests to fail fast. Thanh, perhaps this is a good question for the CircleCI team - is it possible to dynamically enable and disable the resource class / higher container count per-build?